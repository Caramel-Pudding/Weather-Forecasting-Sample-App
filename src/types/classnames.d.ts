declare module "classnames" {
  function classNames(...classes: string[]): string;
  export default classNames;
}
