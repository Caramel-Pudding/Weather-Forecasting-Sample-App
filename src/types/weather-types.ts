export interface City {
  id: number;
  country: string;
  name: string;
}

export interface WeatherData {
  city: City;
}
