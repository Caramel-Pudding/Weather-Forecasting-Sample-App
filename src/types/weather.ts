export enum PartOfTheDay {
  n = "night",
  d = "day",
}

export enum WeatherType {
  Clear = "Clear",
  Rain = "Rain",
  Snow = "Snow",
  Extreme = "Extreme",
}

export interface City {
  id: number;
  country: string;
  name: string;
}

export interface WeatherListItem {
  clouds: {
    all: number;
  };
  dt: number;
  dt_txt: string;
  main: {
    grnd_level: number;
    humidity: number;
    pressure: number;
    sea_level: number;
    temp: number;
    temp_kf: number;
    temp_max: number;
    temp_min: number;
  };
  rain?: {
    "3h"?: number;
  };
  snow?: {
    "3h"?: number;
  };
  sys: {
    pod: PartOfTheDay;
  };
  weather: {
    description: string;
    icon: string;
    id: number;
    main: WeatherType;
  }[];
  wind: {
    deg: number;
    speed: number;
  };
}

export interface WeatherData {
  city: City;
  cnt: number;
  cod: string;
  message: number;
  list: WeatherListItem[];
}
