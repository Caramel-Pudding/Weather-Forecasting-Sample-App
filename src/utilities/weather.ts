import { WeatherListItem } from "@/types/weather";

export const getWeatherListForToday = (
  weatherList: WeatherListItem[]
): WeatherListItem[] => {
  if (!weatherList || weatherList.length === 0) {
    return [];
  }
  const todayDate = new Date(weatherList[0].dt_txt).getDate();
  return weatherList.filter(
    (listItem) => new Date(listItem.dt_txt).getDate() === todayDate
  );
};

export const getTimeStringFromWeatherTimestamp = (
  weatherTimestamp: string
): string => {
  const timestamp = new Date(weatherTimestamp);
  return `${timestamp.getHours()}:00`;
};
