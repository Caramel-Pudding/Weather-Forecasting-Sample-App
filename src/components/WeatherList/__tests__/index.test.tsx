import React from "react";
import { render, cleanup } from "@/tests/test-utils";

import { WeatherListItem } from "@/types/weather";

import { weatherDataStub } from "@/tests/stubs";
import { WeatherList } from "..";

describe("WeatherList Component", () => {
  afterEach(cleanup);

  it("Should render properly if there's proper data", () => {
    // * #TEST: ARRANGE
    const { container } = render(
      <WeatherList
        handleWeatherItemChange={(): void => {}}
        selectedWeatherItemTimestamp={weatherDataStub.list[0].dt}
        weatherItems={weatherDataStub.list as WeatherListItem[]}
      />,
      {}
    );
    // * #TEST: ASSERT
    expect(container).toMatchSnapshot();
  });

  it("Should render amount of buttons equal to amount of incoming list items", () => {
    // * #TEST: ARRANGE
    const { container, getAllByRole } = render(
      <WeatherList
        handleWeatherItemChange={(): void => {}}
        selectedWeatherItemTimestamp={weatherDataStub.list[0].dt}
        weatherItems={weatherDataStub.list as WeatherListItem[]}
      />,
      {}
    );
    const allListButtons = getAllByRole("button");
    // * #TEST: ASSERT
    expect(allListButtons).toHaveLength(36);
    expect(container).toMatchSnapshot();
  });

  it("Should render no buttons if there's no list items", () => {
    // * #TEST: ARRANGE
    const { container, queryAllByRole } = render(
      <WeatherList
        handleWeatherItemChange={(): void => {}}
        selectedWeatherItemTimestamp={weatherDataStub.list[0].dt}
        weatherItems={[] as WeatherListItem[]}
      />,
      {}
    );
    const allListButtons = queryAllByRole("button");
    // * #TEST: ASSERT
    expect(allListButtons).toHaveLength(0);
    expect(container).toMatchSnapshot();
  });
});
