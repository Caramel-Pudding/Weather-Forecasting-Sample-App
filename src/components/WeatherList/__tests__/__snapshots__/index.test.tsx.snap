// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherList Component Should render amount of buttons equal to amount of incoming list items 1`] = `
<div>
  <section
    class="container"
  >
    <ol
      class="weatherItemsRow"
    >
      <li
        class="listItem"
      >
        <button
          class="container containerActive"
          type="button"
        >
          <div
            class="additionalText"
          >
            15:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            14°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            18:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            13°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            21:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            4°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            00:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            0°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            03:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            0°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            06:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            2°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            09:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            3°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            12:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            4°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            15:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            5°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            18:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            3°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            21:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            2°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            00:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            2°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            03:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            2°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            06:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            2°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            09:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            1°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            12:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            2°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            15:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            5°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            18:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            5°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            21:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            1°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            00:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            -2°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            03:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            -4°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            06:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            -5°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            09:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            -6°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            12:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            0°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            15:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            4°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            18:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            5°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            21:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            -1°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            00:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            -4°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            03:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            -4°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            06:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            -5°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            09:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            -5°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            12:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            1°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            15:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            6°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            18:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            6°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            21:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            1°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            00:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            -1°
          </div>
        </button>
      </li>
    </ol>
  </section>
</div>
`;

exports[`WeatherList Component Should render no buttons if there's no list items 1`] = `
<div>
  <section
    class="container"
  >
    <ol
      class="weatherItemsRow"
    />
  </section>
</div>
`;

exports[`WeatherList Component Should render properly if there's proper data 1`] = `
<div>
  <section
    class="container"
  >
    <ol
      class="weatherItemsRow"
    >
      <li
        class="listItem"
      >
        <button
          class="container containerActive"
          type="button"
        >
          <div
            class="additionalText"
          >
            15:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            14°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            18:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            13°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            21:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            4°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            00:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            0°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            03:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            0°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            06:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            2°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            09:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            3°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            12:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            4°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            15:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            5°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            18:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            3°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            21:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            2°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            00:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            2°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            03:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            2°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            06:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            2°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            09:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            1°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            12:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            2°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            15:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            5°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            18:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            5°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            21:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            1°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            00:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            -2°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            03:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            -4°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            06:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            -5°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            09:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            -6°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            12:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            0°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            15:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            4°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            18:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            5°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            21:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            -1°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            00:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            -4°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            03:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            -4°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            06:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            -5°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            09:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            -5°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            12:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            1°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            15:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            6°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            18:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Clear"
              src="icons/weather-sun.svg"
            />
          </section>
          <div
            class="temperature"
          >
            6°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            21:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            1°
          </div>
        </button>
      </li>
      <li
        class="listItem"
      >
        <button
          class="container containerInactive"
          type="button"
        >
          <div
            class="additionalText"
          >
            00:00
          </div>
          <section
            class="imageWrapper"
          >
            <img
              alt="Cloudy"
              src="icons/weather-cloud.svg"
            />
          </section>
          <div
            class="temperature"
          >
            -1°
          </div>
        </button>
      </li>
    </ol>
  </section>
</div>
`;
